sql/core/src/main/scala/org/apache/spark/sql/execution/SortPrefixUtils.scala

```diff
@@ -66,6 +66,29 @@ object SortPrefixUtils {
   }
 
   /**
+   * Returns whether the specified SortOrder can be satisfied with a radix sort on the prefix.
+   */
+  def canSortFullyWithPrefix(sortOrder: SortOrder): Boolean = {
+    sortOrder.dataType match {
+      case BooleanType | ByteType | ShortType | IntegerType | LongType | DateType | TimestampType =>
```

Comment:
Good catch (actually this was causing SortSuite to fail as well). For now I disabled using radix sort for Longs, and added a TODO to implement null-handling.